description.md

```diff
@@ -38,6 +38,71 @@ Put differently, it shows how each content type is to be used in the various tea
 
 ### Assignments
 
+Assignments are practical activities that learners undertake individually or in teams over the span of multiple weeks.
+The primary objective of an assignment is to create a compelling scenario that allows learners to apply the concepts they have learned.
+
+#### Difficulty
+
+An assignment should have a progressive increase in difficulty among its tasks.
+Structuring an assignment with layers of complexity allows students to assess their current knowledge level and offers guidance for improvement.
+
+The level of difficulty should align with the assignment's narrative.
+Presenting a smooth progression of tasks that gradually become more challenging is preferable to artificially labeling them as "easy" or "hard".
+The latter approach can damage the student's experience, as tasks marked as "easy" may offer less sense of accomplishment upon completion, while "hard" tasks might seem intimidating.
+
+Assignments offer the opportunity to explore beyond the studied subject through bonus tasks.
+These additional tasks present challenges that extend the principles applied in other parts of the assignment.
+They can vary from expanding on an existing task to requesting entirely different solutions or providing critical analysis of a particular approach.
+
+Marking these tasks as "bonus" differs from merely labeling their difficulty.
+They are perceived as either alternatives to other tasks or optional.
+However, it is crucial to maintain balance when incorporating bonus tasks into the assignment to ensure students remain engaged enough to tackle them.
+
+#### Grading
+
+All assignments need to incorporate an internal scoring system that tracks the progress of solving each task and assigns a value to it.
+Such a system is crucial because the learning process heavily relies on feedback.
+A manual solution becomes impractical due to scalability issues when dealing with a large number of students and attempts.
```

Comment:
Try to use a less scientific tone as this may be confusing. I think being blunt about it is better: "Manually grading assignments is a time-consuming, repetitive and boring task." -> it implies that the more students you have, the more time you spend.